pkg/limayaml/limayaml.go

```diff
@@ -7,23 +7,23 @@ import (
 )
 
 type LimaYAML struct {
-	Arch         Arch               `yaml:"arch,omitempty"`
-	Images       []File             `yaml:"images"` // REQUIRED
-	CPUs         int                `yaml:"cpus,omitempty"`
-	Memory       string             `yaml:"memory,omitempty"` // go-units.RAMInBytes
-	Disk         string             `yaml:"disk,omitempty"`   // go-units.RAMInBytes
-	Mounts       []Mount            `yaml:"mounts,omitempty"`
-	SSH          SSH                `yaml:"ssh,omitempty"` // REQUIRED (FIXME)
-	Firmware     Firmware           `yaml:"firmware,omitempty"`
-	Video        Video              `yaml:"video,omitempty"`
-	Provision    []Provision        `yaml:"provision,omitempty"`
-	Containerd   Containerd         `yaml:"containerd,omitempty"`
-	Probes       []Probe            `yaml:"probes,omitempty"`
-	PortForwards []PortForward      `yaml:"portForwards,omitempty"`
-	Networks     []Network          `yaml:"networks,omitempty"`
-	Network      NetworkDeprecated  `yaml:"network,omitempty"` // DEPRECATED, use `networks` instead
-	Env          map[string]*string `yaml:"env,omitempty"`     // EXPERIMENAL, see default.yaml
-	DNS          []net.IP           `yaml:"dns,omitempty"`
+	Arch         Arch              `yaml:"arch,omitempty"`
+	Images       []File            `yaml:"images"` // REQUIRED
+	CPUs         int               `yaml:"cpus,omitempty"`
+	Memory       string            `yaml:"memory,omitempty"` // go-units.RAMInBytes
+	Disk         string            `yaml:"disk,omitempty"`   // go-units.RAMInBytes
+	Mounts       []Mount           `yaml:"mounts,omitempty"`
+	SSH          SSH               `yaml:"ssh,omitempty"` // REQUIRED (FIXME)
+	Firmware     Firmware          `yaml:"firmware,omitempty"`
+	Video        Video             `yaml:"video,omitempty"`
+	Provision    []Provision       `yaml:"provision,omitempty"`
+	Containerd   Containerd        `yaml:"containerd,omitempty"`
+	Probes       []Probe           `yaml:"probes,omitempty"`
+	PortForwards []PortForward     `yaml:"portForwards,omitempty"`
+	Networks     []Network         `yaml:"networks,omitempty"`
+	Network      NetworkDeprecated `yaml:"network,omitempty"` // DEPRECATED, use `networks` instead
+	Env          map[string]string `yaml:"env,omitempty"`
```

Comment:
Why was *string changed to string?

Reply:
There was some reason, but I don't remember, and it no longer applies; I can easily change it back. Please confirm if you want me to do that!

I asked back in https://github.com/lima-vm/lima/pull/195#discussion_r703894040 why it was a `*string` to begin with because I was having issues with them in templates elsewhere, but didn't receive a response.

So I'm still curious why it should be a `*string`. Theoretically we could use `nil` vs `""` to "unset" an environment variable, but that doesn't make much sense in `/etc/environment`, so I still have no idea...

Reply:
So it doesn't matter because (1) we no longer pass `y.Env` directly to the template, and (2) because we don't use `printf` on the values, but if we did, here is the difference (using this template):

```yaml
#LIMA-START
{{- range $key, $val := .Env}}
{{$key}}={{$val}}
{{printf "%s=%q" $key $val}}
{{- end}}
#LIMA-END
```

produces this output:

```
#LIMA-START
KEY=value"value
KEY=%!q(*string=0xc0001bb2a0)
#LIMA-END
```

which is what prejudices me against the `*string`. 😸